### PR TITLE
Fixes AIs being able to control airlocks without wires

### PR DESCRIPTION
**Pull request by ShiftyRail**
_Not tested yet!_

## Reasoning

Previously if you had cut an AI  Control wire, but it had already open the window, this changed nothing because the Topic() call did not check if the wire was intact when sending the signal.
This fixes that.

:cl:
- bugix: AIs are no longer able to control an airlock whose AI Control Wire had been cut.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
A bit misleading, there's still the period of time until it takes to take control of a cut wire door

**Comment by Pathid** [comment]
~~That changelog could stand to be more clear. It makes it so they can't bypass the hack time by having the window already open, that's all.~~
EDIT: Wait a minute. You are doing _**that**_ aren't you?

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -685,11 +685,29 @@ About the new airlock wires panel:
...
+
 /obj/machinery/door/airlock/Topic(href, href_list, var/nowindow = 0)
 	// If you add an if(..()) check you must first remove the var/nowindow parameter.
 	// Otherwise it will runtime with this kind of error: null.Topic()
 	var/turf/T = get_turf(usr)
-	if(!isAI(usr) && T.z != z)
+
+	var/user_is_AI = isAI(usr)
+
+	if(!user_is_AI && T.z != z)
```
Comment (MadmanMartian) [inline code/game/machinery/doors/airlock.dm:710]: >this whole block

what am I missing here?

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -705,14 +723,14 @@ About the new airlock wires panel:
 			return
 
 	if(isAdminGhost(usr) || (istype(usr, /mob/living/silicon) && src.canAIControl() && operating != -1))
+		if (user_is_AI && src.isWireCut(AIRLOCK_WIRE_AI_CONTROL))
```
Comment (Pathid) [inline code/game/machinery/doors/airlock.dm:726]: Isn't this defeating the purpose of ``canAIControl()``?

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/game/machinery/doors/airlock.dm:710]: Remote signalers do not work accross z-levels

**Comment by ShiftyRail** [comment]
Well, it looks like I've been completely misunderstood about how the mechanics of it work
What I want to do is therefore not a bugfix but a balance PR, mainly about why the AI control wire is just a slight delay for the big silicon overlord rather than a reliable way to block it

**Pull request closed without merging**